src/services/codefixes/helpers.ts

```diff
@@ -247,41 +247,70 @@ namespace ts.codefix {
         );
     }
 
-    export function createMethodFromCallExpression(
+    export function createFunctionFromCallExpression<T extends FunctionDeclaration | MethodDeclaration>(

```

Comment:
The type parameter and the `kind` look like they're only ever both `MethodDeclaration` or `FunctionDeclaration`, so this seems redundant. Seems like you could skip on the kind if you want to go the type parameter route or just do the type assertion / check afterward.